src/portable/st/synopsys/dcd_synopsys.c

```diff
@@ -550,14 +550,17 @@ static void transmit_fifo_packet(uint8_t fifo_num, uint8_t * src, uint16_t len){
 	}
 
 	// Write the remaining 1-3 bytes into fifo
-	uint32_t tmp_word = 0;
-	switch(len & 0x0003){
-		case 3: tmp_word |= src[2] << 16;
-		case 2: tmp_word |= src[1] << 8;
-		case 1: tmp_word |= src[0];
-			*tx_fifo = tmp_word;
-			break;
-		default: break;
+	uint8_t bytes_rem = len & 0x03;
```

Comment:
Thanks for this information. I was a little shocked when I created this PR and suddenly all builds on GitHub failed while my toolchain did not complain about this. Would you prefer the if or previous switch solution?

Reply:
I am fine with either way, though I prefer the **simple if on single line**, it will be even clearer than swtich with fallthru comment. Hmm, which gcc you are on, it should be caught when compiling locally on your machine.

Reply:
I'm currently working with **GNU Tools for STM32**, which is a patched version of _GNU Tools for ARM Embedded Processors 7-2018q2-update Release_. I guess it is only some default properties they changed here and that's why I do not see a warning on my configuration. To bad the patch list they are referencing on their website is not available anymore.

Reply:
That is too old, the current gcc is v9 already. There is a major update/upgrade regarding the link time optimization from v7 to v8. It also enforce lots more warnings as well.